cli/js/web/fetch/multipart.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+
+import { Buffer } from "../../buffer.ts";
+import { bytesSymbol } from "../blob.ts";
+import { DomFileImpl } from "../dom_file.ts";
+import { TextEncoder } from "../text_encoding.ts";
+
+const encoder = new TextEncoder();
+
+export class MultipartBuilder {
+  readonly boundary: string;
+  readonly formData: FormData;
+  readonly writer: Buffer;
+  constructor(formData: FormData, boundary?: string) {
+    this.boundary = boundary ?? this.#createBoundary();
+    this.formData = formData;
+    this.writer = new Buffer();
+  }
+
+  getContentType(): string {
+    return `multipart/form-data; boundary=${this.boundary}`;
+  }
+
+  getBody(): Uint8Array {
+    for (const [fieldName, fieldValue] of this.formData.entries()) {
+      if (fieldValue instanceof DomFileImpl) {
+        this.#writeFile(fieldName, fieldValue);
+      } else this.#writeField(fieldName, fieldValue as string);
+    }
+
+    this.writer.writeSync(encoder.encode(`\r\n--${this.boundary}--`));
```

Comment:
I used `writeSync` since in `master` we're building it synchronous too, and in any case, `Buffer.write` also uses `.writeSync` internally.

If someone prefers `.write` instead, I'll do the change, don't have strong opinions here.

Reply:
That's fine 👍